discord/components.py

```diff
@@ -1347,6 +1348,72 @@ def walk_components(self) -> Iterator[Component]:
         yield from [self.component]
 
 
+class FileUpload(Component):
+    """Represents an File Upload field from the Discord Bot UI Kit.
+
+    This inherits from :class:`Component`.
+
+    .. note::
+
+        This class is not useable by end-users; see :class:`discord.ui.FileUpload` instead.
+
+    .. versionadded:: 2.7
+
+    Attributes
+    ----------
+    custom_id: Optional[:class:`str`]
+        The custom ID of the file upload field that gets received during an interaction.
+    min_values: Optional[:class:`int`]
+        The minimum number of files that must be uploaded.
+        Defaults to 0.
+    max_values: Optional[:class:`int`]
+        The maximum number of files that can be uploaded.
+    required: Optional[:class:`bool`]
+        Whether the file upload field is required or not. Defaults to `True`.
+    id: Optional[:class:`int`]
+        The file upload's ID.
+    """
+
+    __slots__: tuple[str, ...] = (
+        "type",
+        "custom_id",
+        "min_values",
+        "max_values",
+        "required",
+        "id",
+    )
+
+    __repr_info__: ClassVar[tuple[str, ...]] = __slots__
+    versions: tuple[int, ...] = (1, 2)
+
+    def __init__(self, data: FileUploadComponentPayload):
+        self.type = ComponentType.file_upload
+        self.id: int | None = data.get("id")
+        self.custom_id = data["custom_id"]
+        self.min_values: int | None = data.get("min_values", None)
+        self.max_values: int | None = data.get("max_values", None)
+        self.required: bool = data.get("required", True)
+
+    def to_dict(self) -> FileUploadComponentPayload:
+        payload = {
+            "type": 19,
+            "custom_id": self.custom_id,
+        }
+        if self.id is not None:
+            payload["id"] = self.id
+
+        if self.min_values:
+            payload["min_values"] = self.min_values
+
+        if self.max_values:
+            payload["max_values"] = self.max_values
```

Comment:
```suggestion
        if self.min_values is not None:
            payload["min_values"] = self.min_values

        if self.max_values is not None:
            payload["max_values"] = self.max_values
```
`0` is valid, at least for min_values.